state/src/storage/tx/matching_pools.rs

```diff
@@ -0,0 +1,206 @@
+//! Helpers for accessing information about matching pools in the database
+
+use common::types::wallet::OrderIdentifier;
+use libmdbx::{TransactionKind, RW};
+
+use crate::{storage::error::StorageError, POOL_TABLE};
+
+use super::StateTxn;
+
+// -------------
+// | Constants |
+// -------------
+
+/// The name of the global matching pool
+pub const GLOBAL_MATCHING_POOL: &str = "global";
+
+/// The prefix of all matching pool keys
+pub const POOL_KEY_PREFIX: &str = "matching-pool/";
+
+/// The error message used when trying to create a pool that already exists
+const POOL_EXISTS_ERR: &str = "pool already exists";
+/// The error message used when allocating an order to a nonexistent pool
+const POOL_DOES_NOT_EXIST_ERR: &str = "pool does not exist";
+/// The error message used when a non-empty pool is attempted to be destroyed
+const POOL_NOT_EMPTY_ERR: &str = "pool not empty";
+
+// ---------------
+// | Key Helpers |
+// ---------------
+
+/// The key for the set of all matching pools
+pub fn all_pools_key() -> String {
+    "all-pools".to_string()
+}
+
+/// The key for the given order's matching pool
+pub fn matching_pool_key(order_id: &OrderIdentifier) -> String {
+    format!("{}{}", POOL_KEY_PREFIX, order_id)
+}
+
+// -----------
+// | Getters |
+// -----------
+
+impl<'db, T: TransactionKind> StateTxn<'db, T> {
+    /// Get the name of the matching pool the given order is in, if it's been
+    /// allocated to one
+    pub fn get_pool_for_order(
```

Comment:
can we do `:s/pool/matching_pool`? The naming is fair in the context of this file, but when I look at the documentation of the `StateTxn` object, i see "pool" thrown around in signatures, and it's unclear what the "pool" is. It's clearer if these are labelled `matching_pool`. Things like `pool_exists` are underspecified for example, so i propose referring to pools as matching pools across the interface.